app/scripts/controllers/metamask-notifications/services/onchain-notifications.ts

```diff
@@ -0,0 +1,288 @@
+): Promise<UserStorage> {
+  const triggersToDelete = uuids.map((uuid) => ({
+    id: uuid,
+    token: createSHA256Hash(uuid + storageKey),
+  }));
+
+  try {
+    const response = await makeApiCall(
+      bearerToken,
+      TRIGGER_API_BATCH_ENDPOINT,
+      'DELETE',
+      triggersToDelete,
+    );
+
+    if (!response.ok) {
+      throw new Error(
+        `Failed to delete on-chain notifications for uuids ${uuids.join(', ')}`,
+      );
+    }
+
+    // Update the state of the deleted trigger to false
+    for (const uuid of uuids) {
+      for (const address in userStorage) {
+        if (Object.hasOwn(userStorage, address)) {
+          for (const chainId in userStorage[address]) {
+            if (userStorage?.[address]?.[chainId]?.[uuid]) {
+              delete userStorage[address][chainId][uuid];
+            }
+          }
+        }
+      }
+    }
+
+    // Follow-up cleanup, if an address had no triggers whatsoever, then we can delete the address
+    const isEmpty = (obj = {}) => Object.keys(obj).length === 0;
+    for (const address in userStorage) {
+      if (Object.hasOwn(userStorage, address)) {
+        for (const chainId in userStorage[address]) {
+          // Chain isEmpty Check
+          if (isEmpty(userStorage?.[address]?.[chainId])) {
+            delete userStorage[address][chainId];
+          }
+        }
+
+        // Address isEmpty Check
+        if (isEmpty(userStorage?.[address])) {
+          delete userStorage[address];
+        }
+      }
+    }
+  } catch (err) {
+    log.error(
+      `Error deleting on-chain notifications for uuids ${uuids.join(', ')}:`,
+      err,
+    );
+    throw err;
+  }
+
+  return userStorage;
+}
+
+/**
+ * Fetches on-chain notifications for the given user storage and BearerToken.
+ * This method iterates through the userStorage to find enabled triggers and fetches notifications for those triggers.
+ * It makes paginated API calls to the notifications service, transforming and aggregating the notifications into a single array.
+ * The process stops either when all pages have been fetched or when a page has less than 100 notifications, indicating the end of the data.
+ *
+ * @param userStorage - The user's storage object containing trigger information.
+ * @param bearerToken - The JSON Web Token used for authentication in the API call.
+ * @returns A promise that resolves to an array of OnChainRawNotification objects. If no triggers are enabled or an error occurs, it may return an empty array.
+ */
+export async function getOnChainNotifications(
+  userStorage: UserStorage,
+  bearerToken: string,
+): Promise<OnChainRawNotification[]> {
+  const triggerIds = traverseUserStorageTriggers(userStorage, {
+    mapTrigger: (t) => {
+      if (!t.enabled) {
+        return undefined;
+      }
+      return t.id;
+    },
+  });
+
+  if (triggerIds.length === 0) {
+    return [];
+  }
+
+  const onChainNotifications: OnChainRawNotification[] = [];
+  const PAGE_LIMIT = 2;
```

Comment:
OK IDM pagination at controller level. We can tweak this once we connect the UI.

If a page = 100 on-chain notifications then we will have:
```
+ 200 On-Chain
+ X Off-Chain (aka feature announcement)
+ Y Snap notifications
---
Alot of notifications
```

It might not be reasonable to show this many notifications. Lets update once we integrate UI.